gpjax/kernels/stationary/changepoint.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+"""
+AUTHOR: Jake Yeung (jakeyeung@gmail.com)
+CREATED ON:  2023-07-28
+LAST CHANGE: see git log
+LICENSE:     Apache License
+"""
+
+from dataclasses import dataclass
+
+from beartype.typing import Union
+import jax
+import jax.numpy as jnp
+from jaxtyping import Float
+import tensorflow_probability.substrates.jax.bijectors as tfb
+
+import gpjax as gpx
+from gpjax.base.param import param_field
+from gpjax.typing import (
+    Array,
+    ScalarFloat,
+)
+
+
+@dataclass
+class ChangePointRBF(gpx.kernels.AbstractKernel):
+    r"""The change point kernel for two RBFs."""
+
+    variance1: ScalarFloat = param_field(
+        jnp.array(1.0), bijector=tfb.Softplus(), trainable=True
+    )
+    lengthscale1: Union[ScalarFloat, Float[Array, " D"]] = param_field(
+        jnp.array(1.0), bijector=tfb.Softplus(), trainable=True
+    )
+    variance2: ScalarFloat = param_field(
+        jnp.array(1.0), bijector=tfb.Softplus(), trainable=True
+    )
+    lengthscale2: Union[ScalarFloat, Float[Array, " D"]] = param_field(
+        jnp.array(1.0), bijector=tfb.Softplus(), trainable=True
+    )
+    tswitch: float = param_field(
+        jnp.float64(1.0), bijector=tfb.Identity(), trainable=False
+    )
+    name: str = "ChangePoint"
+
+    def __call__(self, x: Float[Array, " D"], y: Float[Array, " D"]) -> ScalarFloat:
+        r"""Compute the changepoint kernel for two RBFs between a pair of arrays.
+        ```
+
+        Args:
+            x (Float[Array, " N"]): The left hand argument of the kernel function's call.
+            y (Float[Array, " N"]): The right hand argument of the kernel function's call.
+            self.tswitch: The point at which to change to a different kernel.
+            self.variance1: Variance parameter for first RBF (if x, y both less than tswitch)
+            self.lengthscale1: Lengthscale parameter for first RBF (if x, y both less than tswitch)
+            self.variance2: Variance parameter for second RBF (if x, y both greater than or equal to tswitch)
+            self.lengthscale2: Lengthscale parameter for second RBF (if x, y both greater than or equal to tswitch)
```

Comment:
These lines should be move to the objects initialisation.

Reply:
It's now moved, and simplified to just have a list of kernels and tswitch.